File: Chapter 05/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter05/section_5_4_server_monitor/after/infrastructure/adapters/ConsoleAdapter.java
package com.grokkingsoftwarearchitecture.chapter05.section_5_4_server_monitor.after.infrastructure.adapters;

import com.grokkingsoftwarearchitecture.chapter05.section_5_4_server_monitor.after.core.ports.AlertPort;

/**
 * ADAPTER 2: The "Dev" Adapter.
 * This adapter is perfect for local development. It proves that 
 * the Core doesn't care if the alert goes to a multi-million dollar cloud 
 * messaging service or simply prints to the local screen.
 */
public class ConsoleAdapter implements AlertPort {

    // ANSI escape codes to mimic C#'s Console.ForegroundColor
    private static final String ANSI_RED = "\u001B[31m";
    private static final String ANSI_RESET = "\u001B[0m";

    @Override
    public void sendAlert(String message) {
        // We wrap the message in the ANSI codes to print it in red,
        // then immediately reset the color so we don't bleed into other logs.
        System.out.println(ANSI_RED + "(DEV ADAPTER) ALERT: " + message + ANSI_RESET);
    }
}

File: Chapter 05/C#/Section_5_4_Server_Monitor/After/Infrastructure/Adapters/ConsoleAdapter.cs
using Chapter05.ServerMonitor.After.Core.Ports;
using System;

namespace Chapter05.ServerMonitor.After.Infrastructure.Adapters
{
    /// <summary>
    /// ADAPTER 2: The "Dev" Adapter.
    /// Proves that the Core logic remains identical regardless of the 
    /// infrastructure being used (Cloud SMS vs. Local Screen).
    /// </summary>
    public class ConsoleAdapter : IAlertPort
    {
        /// <summary>
        /// Sends an alert to the local console window.
        /// </summary>
        public void SendAlert(string message)
        {
            // Visual feedback for development to help developers track the state.
            Console.ForegroundColor = ConsoleColor.Red;
            Console.WriteLine($"(DEV ADAPTER) ALERT: {message}");
            Console.ResetColor();
        }
    }
}